cmd/oras/push.go

```diff
@@ -136,33 +139,45 @@ func runPush(opts pushOptions) error {
 	return opts.ExportManifest(ctx, store, desc)
 }
 
-func packManifest(ctx context.Context, store *file.Store, annotations map[string]map[string]string, opts *pushOptions) (ocispec.Descriptor, error) {
+func packManifest(ctx context.Context, store *file.Store, annotations map[string]map[string]string, opts *pushOptions) (manifestDesc ocispec.Descriptor, err error) {
 	var packOpts oras.PackOptions
 	packOpts.ConfigAnnotations = annotations[annotationConfig]
 	packOpts.ManifestAnnotations = annotations[annotationManifest]
 
-	if opts.artifactType != "" {
-		packOpts.ConfigMediaType = opts.artifactType
-	}
-	if opts.manifestConfigRef != "" {
-		path, mediatype := parseFileReference(opts.manifestConfigRef, oras.MediaTypeUnknownConfig)
-		desc, err := store.Add(ctx, annotationConfig, mediatype, path)
+	if len(opts.FileRefs) != 0 {
+		// pack OCI artifact
+		if opts.artifactType != "" {
+			packOpts.ConfigMediaType = opts.artifactType
+		}
+		if opts.manifestConfigRef != "" {
+			path, mediatype := parseFileReference(opts.manifestConfigRef, oras.MediaTypeUnknownConfig)
+			desc, err := store.Add(ctx, annotationConfig, mediatype, path)
+			if err != nil {
+				return ocispec.Descriptor{}, err
+			}
+			desc.Annotations = packOpts.ConfigAnnotations
+			packOpts.ConfigDescriptor = &desc
+		}
+		descs, err := loadFiles(ctx, store, annotations, opts.FileRefs, opts.Verbose)
+		if err != nil {
+			return ocispec.Descriptor{}, err
+		}
+		manifestDesc, err = oras.Pack(ctx, store, descs, packOpts)
+		if err != nil {
+			return ocispec.Descriptor{}, err
+		}
```

Comment:
Nit: this can be merged with the same error checking code in else block. You can put the merged error checking code right after the if-else.